graphql_compiler/schema/sqlalchemy_schema.py

```diff
@@ -1,17 +1,115 @@
 # Copyright 2019-present Kensho Technologies, LLC.
 from collections import namedtuple
+import sqlalchemy
+import six
+
+from . import is_vertex_field_name
+
+from graphql import GraphQLList
+from graphql.type.definition import GraphQLInterfaceType, GraphQLObjectType, GraphQLUnionType
 
 
 # Complete schema information sufficient to compile GraphQL queries to SQLAlchemy
 #
-# It describes the tables that correspond to each type, and gives instructions on how
-# to perform joins for each vertex field. The property fields on each type are implicitly
-# mapped to columns with the same name on the corresponding table.
+# It describes the tables that correspond to each type (object type, interface type or union type),
+# and gives instructions on how to perform joins for each vertex field. The property fields on each
+# type are implicitly mapped to columns with the same name on the corresponding table.
+#
+# NOTES:
+# - RootSchemaQuery is a special type that does not need a corresponding table.
+# - Builtin types like __Schema, __Type, etc. don't need corresponding tables.
+# - Builtin fields like _x_count do not need corresponding columns.
 SQLAlchemySchemaInfo = namedtuple('SQLAlchemySchemaInfo', (
-    'schema',     # GraphQLSchema
-    'tables',     # dict mapping every graphql object type name in the schema to a sqlalchemy table
-    'junctions',  # dict mapping every graphql object type name in the schema to:
-                  #    dict mapping every vertex field name at that type to a dict with keys:
-                  #        from_column_name: string, column name to join from
-                  #        to_column_name: string, column name to join to
+    # GraphQLSchema
+    'schema',
+
+    # dict mapping every graphql object type or interface type name in the schema to
+    # a sqlalchemy table
+    'tables',
+
+    # dict mapping every graphql object type or interface type name in the schema to:
+    #    dict mapping every vertex field name at that type to a dict with keys:
+    #        from_column_name: string, column name to join from
+    #        to_column_name: string, column name to join to
+    'junctions',
+
+    # dict mapping every graphql object type or interface type name in the schema to:
+    #    dict mapping every list-valued property field name to a dict with keys:
+    #        junction_name: string, previously specified junction that provides the set contents
+    #        value_column: string, the column in the junction target table that has the elements
+    'set_valued_fields',
```

Comment:
In Postgres, you can use `sqlalchemy.ARRAY` instead of this. I'll clarify this when I add the SQL flavor to the `SQLAlchemySchemInfo`